target/linux/ramips/dts/mt7621_dlink_dir-878-a1.dts

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "dlink,dir-878-a1", "mediatek,mt7621-soc";
+	model = "D-Link DIR-878 A1";
+
+	aliases {
+		led-boot = &led_power_green;
+		led-failsafe = &led_power_green;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_green;
+		label-mac-device = &ethernet;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power_orange {
+			label = "dir-878-a1:orange:power";
+			gpios = <&gpio0 8 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_green: power_green {
+			label = "dir-878-a1:green:power";
+			gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
+		};
+
+		net_orange {
+			label = "dir-878-a1:orange:net";
+			gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;
+		};
+
+		net_green {
+			label = "dir-878-a1:green:net";
+			gpios = <&gpio0 3 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 15 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+		
+		wps {
+			label = "wps";
+			gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+		
+		wifi {
+			label = "wifi";
+			gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RFKILL>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+		m25p,chunked-io = <32>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x20000>;
+				read-only;
+			};
+
+			partition@60000 {
+				compatible = "ubootpad96,uimage";
+				label = "firmware";
+				reg = <0x60000 0xfa0000>;
+			};
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+		ieee80211-freq-limit = <2400000 2500000>;
+
+		led {
+			led-active-low;
+		};
+	};
+};
+
+&pcie1 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+
+		led {
+			led-active-low;
+		};
+	};
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0xe000>;
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c","wdt","uart2", "uart3", "rgmii2", "sdhci";

```

Comment:
pingroup `rgmii2` (GPIO 22-33), `uart2` (GPIO 9-12) and `sdhci` (GPIO 41-48) are not used and can be dropped.
GPIO 16 is in `jtag` pingroup.

Should be changed to `ralink,group = "i2c", "uart3", "jtag", "wdt";`

Reply:
Can you explain me how to deduce that (so, what you just did) and/or refer to some literature on it?

Reply:
Okay, found it out myself: I just have to look for gpio "numbers" used in the DT and then look up the assignment in the corresponding data sheet (mt7621 data sheet in this case)....

Reply:
@Lucky1openwrt

state_default is already set up in mt7621.dtsi. So, you can remove the parent &pinctrl and just put the following in your file:

```
&state_default {
   gpio {
      ...
   };
};
```

Reply:
ok changed

Reply:
I think something I did here makes the board noticeable unstable on cold boot
I noticed a few times in the logs I think it followed after setting up the switch and or vlan
I have suspicions of a GPIO pin connected to a reset on the switch or something  
I will look into this more I can't repeat the fault while the board is warm

Reply:
ok it was the last kernel: bump 4.14 to 4.14.158 that broke it
https://git.openwrt.org/?p=openwrt/openwrt.git;a=commitdiff;h=d395583d697c17f00a533d6efa13754c21630eda